File: src/GuessNumberHigherOrLower.java
public class GuessNumberHigherOrLower {
    public int guessNumber(int n) {
        int left = 1, right = n, mid;
        while (left <= right) {
            mid = left + (right - left) / 2;
            int guess = guess(mid);
            if (guess == 0) return mid;
            else if (guess == -1) right = mid - 1;
            else left = mid + 1;
        }
        return -1;
    }

    // dummy function - has been defined by Leetcode
    public int guess(int num) {
        return 0;
    }
}

File: python/guess_number_higher_or_lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0

def guess(num: int) -> int:
    # temp purposes
    return 0


class Solution:
    def guessNumber(self, n: int) -> int:
        left, right = 1, n
        while left <= right:
            middle = (left + right) // 2
            hypothesis = guess(middle)
            if hypothesis == 0:
                return middle
            elif hypothesis == 1:
                left = middle + 1
            else:
                right = middle - 1
        return 0
